bin/syclcc

```diff
@@ -222,8 +226,62 @@ class source_file_processor:
 
 cpp_files = set([".cpp",".cxx",".c++",".cc",".c"])
 
-## Compiler backend using nvcc
-class cuda_compiler:
+## Compiler backend for CUDA using Clang
+class cuda_clang_compiler:
+  def __init__(self,config):
+    self.device_compiler = config.cuda_clang_compiler
+    self.config = config
+    self.is_clang_available = is_program_executable(self.device_compiler)
+
+  def includes_linking_stage(self, args):
+    for arg in args:
+      if (arg.startswith("-E") or
+          arg.startswith("-fsyntax-only") or
+          arg.startswith("-S") or
+          arg.startswith("-c")):
+        return False
+    return True
+
+  def run(self, args):
+    source_processor = source_file_processor(self.config, ".cu", args)
+    result = None
+    try:
+      transformed_args = []
+      for arg in args:
+        if is_source_file(arg):
+          transformed_args.append(source_processor.process_file(arg))
+        else:
+          transformed_args.append(arg)
+
+      compiler_args = ["--cuda-gpu-arch=sm_52",
+                        "-std=c++14",
+                        "-pthread",
+                        "-Wno-unused-command-line-argument",
+                        "-Wno-deprecated-declarations"]
+
+      compiler_args += self.config.hipsycl_common_arguments
+      compiler_args += transformed_args
+
+      # To avoid warnings only add linker flags if this run will invoke the linker
+      # TODO: This mechanism could also be used to conditionally link libhipSYCL
+      if self.includes_linking_stage(args):
+        compiler_args += ["-lcudart_static",
+                           "-ldl",
+                           "-lrt"]
```

Comment:
We should also conditionally link against libhipSYCL as you propose, since this is the behavior of the other compiler classes in `syclcc`.

Reply:
Sorry, forget that, this is of course handled by `config.hipsycl_common_arguments` :)

Reply:
I thought maybe we could at some point extract some of the common functionality that all compilers share into a base class which could then also handle stuff like this.

Reply:
Yes, that makes sense. The current approach was implemented when things were still very simple with only two backends around ;)